stubs/invoke/invoke/parser/parser.pyi

```diff
@@ -0,0 +1,40 @@
+from typing import Any
+
+def is_flag(value): ...
+def is_long_flag(value): ...
+
+class Parser:
+    initial: Any
+    contexts: Any
+    ignore_unknown: Any
+    def __init__(self, contexts=..., initial=..., ignore_unknown: bool = ...) -> None: ...
+    def parse_argv(self, argv): ...
+
+class ParseMachine:
+    initial_state: str
+    def changing_state(self, from_, to) -> None: ...
+    ignore_unknown: Any
+    initial: Any
+    flag: Any
+    flag_got_value: bool
+    result: Any
+    contexts: Any
+    def __init__(self, initial, contexts, ignore_unknown) -> None: ...
+    @property
+    def waiting_for_flag_value(self): ...
+    def handle(self, token) -> None: ...
+    def store_only(self, token) -> None: ...
+    def complete_context(self) -> None: ...
+    context: Any
+    def switch_to_context(self, name) -> None: ...
+    def complete_flag(self) -> None: ...
+    def check_ambiguity(self, value): ...
+    def switch_to_flag(self, flag, inverse: bool = ...) -> None: ...
+    def see_value(self, value) -> None: ...
+    def see_positional_arg(self, value) -> None: ...
+    def error(self, msg) -> None: ...
+
+class ParseResult(list[ParseResult]):
```

Comment:
The docstring says "A `.ParseResult` (a ``list`` subclass containing some number of `.ParserContext` objects).", so the base class here seems wrong.

Reply:
So what I see is `List-like object with some extra parse-related attributes.`:
https://github.com/pyinvoke/invoke/blob/ed94c59f1eacc700dac3815530493e0809fe001d/invoke/parser/parser.py#L426

--> a class extending list --> how can I type that?

Reply:
It should have `list[ParserContext]` as the base class, since it extends list and the list contains ParserContext instances.